cmd/tenant/main.go

```diff
@@ -328,6 +352,135 @@ func reindex(c *cli.Context) (err error) {
 	return nil
 }
 
+func cleanup(c *cli.Context) (err error) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	dry := c.Bool("dry-run")
+
+	// Load the organizations from the CSV file
+	var f *os.File
+	if f, err = os.Open(c.String("orgs")); err != nil {
+		return cli.Exit(err, 1)
+	}
+
+	// Ensure there is a header row
+	var header []string
+	reader := csv.NewReader(f)
+	if header, err = reader.Read(); err != nil {
+		return cli.Exit(err, 1)
+	}
+
+	// Find the ID column or assume the first column is the ID
+	var idCol int
+	for i, col := range header {
+		if strings.ToLower(col) == "id" {
+			idCol = i
+			break
+		}
+	}
+
+	orgs := make(map[ulid.ULID]struct{})
+	for {
+		var record []string
+		if record, err = reader.Read(); err != nil {
+			break
+		}
+
+		var id ulid.ULID
+		if id, err = ulid.Parse(record[idCol]); err != nil {
+			return cli.Exit(fmt.Errorf("could not parse org ID: %s", record[idCol]), 1)
+		}
+
+		orgs[id] = struct{}{}
+	}
+
+	if len(orgs) == 0 {
+		return cli.Exit(fmt.Errorf("no organizations found in CSV file"), 1)
```

Comment:
Hopefully this helps guard against accidentally deleting tenants and members (along with the dry run option and the orgs file being a required argument).